### PR TITLE
Pro 2223 batch operations modal

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-2223](https://linear.app/apostrophecms/issue/PRO-2223/as-an-editor-i-can-confirm-starting-a-batch-operation-after-clicking)

## Purpose
Opening confirmation modal when clicking on batch operation button, uses the informations from the `modalOptions` object of the batch operation. Also works with grouped operations. Count and label are passed to all translation keys passed from the batch operations config, since this is the only dynamic data needed.

## How to test
Open a piece module modal, click on the archive button, it should open a modal with the text passed from the batch operations config. You can test with grouped ones, to do that add `archive` to the group config.

### CODE REVIEW

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue [hunk 1]
```diff
@@ -101,23 +91,23 @@ export default {
...
-    batchOperations: {
-      type: Array,
-      default: () => []
+      default: () => {}
     },
     displayedItems: {
       type: Number,
       required: true
+    },
+    checkedCount: {
```
Comment (abea) [inline modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:104]: Some of these required properties are throwing warnings in the media manager modal.

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/i18n/i18n/en.json [hunk 2]
```diff
@@ -17,6 +17,8 @@
   "archiveTypeAffirmativeLabel": "Yes, archive {{ type }}",
   "archiveTypeNote": "You are currently viewing the {{ type }} you want to archive. When it is archived you will be returned to the home page.",
   "archived": "Archived",
+  "archivingBatchConfirmation": "Are you sure you want to archive {{ count }} {{ type }}?",
```
[the remark this answers is not in the document]
Comment (ValJed) [reply modules/@apostrophecms/i18n/i18n/en.json:20]: I just added them when merging with your work. They are below.

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue [hunk 3]
```diff
@@ -211,11 +202,37 @@ export default {
...
-      this.$emit('batch', action);
-    },
     registerPageChange(pageNum) {
       this.$emit('page-change', pageNum);
+    },
+    async operationModal ({
+      modalOptions, action, operations
+    }) {
+      const {
+        title, description, form
```
[the remark this answers is not in the document]
Comment (ValJed) [reply modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:212]: Could idead, I also forgot to use the confirmation button, fixed.

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue [hunk 4]
```diff
@@ -211,11 +202,37 @@ export default {
...
+      } = action && operations
+        ? (operations.find((op) => op.action === action)).modalOptions
+        : modalOptions;
+
+      const interpolations = {
+        count: this.checkedCount,
+        type: this.checkedCount === 1
+          ? this.moduleOptions.label
```
[the remark this answers is not in the document]
Comment (ValJed) [reply modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:220]: Oh right

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue [hunk 5]
```diff
@@ -211,11 +202,37 @@ export default {
...
+          : this.moduleOptions.pluralLabel
+      };
+
+      // TODO: request batch operation to backend
```
[the remark this answers is not in the document]
Comment (ValJed) [reply modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:224]: Removed.

File: modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue (hunk 1 above)
Comment (ValJed) [reply modules/@apostrophecms/modal/ui/apos/components/AposDocsManagerToolbar.vue:112]: I cleaned props and fixed these errors, nicely spotted.

**Review by abea — APPROVED** [review]

**Comment by abea** [comment]
Passed test manually. npm audit prevents Circle from passing.

**Pull request merged**